### PR TITLE
Feat: Authentication

### PR DESCRIPTION
**Pull request by munjalSaksham71**
Feat: Authentication for regal play

### CODE REVIEW

**Review by Avin008 — APPROVED** [review]
Everything looks good. good to merge

File: src/App.js [hunk 1]
```diff
@@ -3,13 +3,20 @@ import Header from "./components/Header/Header";
 import LandingPage from "./Screen/LandingPage/LandingPage";
 import { Routes, Route } from "react-router-dom";
 import VideoListing from "./Screen/VideoListing/VideoListing";
+import MockMan from "mockman-js";
+import LoginScreen from "./Screen/LoginScreen/LoginScreen";
+import SignupScreen from "./Screen/SignupScreen/SignupScreen";
```
Comment (tusharkandpal) [inline src/App.js]: can use barrel exports

File: src/Screen/LoginScreen/LoginScreen.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    const guestLoginHandler = async (e) => {
+      e.preventDefault();
+      try {
+          await loginUser("adarshbalika@gmail.com", "adarshBalika123");
+      } catch (error) {
+          alert("Something went wrong.")
+      }
+    }
```
Comment (tusharkandpal) [inline src/Screen/LoginScreen/LoginScreen.jsx:32]: you can give a common `loginHandler` in auth-context for the same by just passing the email & password as arguments.

File: src/Screen/SignupScreen/SignupScreen.css [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+.form-height {
+    width: 35%;
+}
+ 
+button, input {
+    background: var(--text-primary);
+    color: var(--background-a);
+}
```
Comment (tusharkandpal) [inline src/Screen/SignupScreen/SignupScreen.css]: can give descendants combinator to be more specific with this style.

File: src/index.js [hunk 4]
```diff
@@ -14,7 +13,9 @@ ReactDOM.render(
   <BrowserRouter>
     <VideoContextProvider>
       <FilterContextProvider>
-        <App />
+        <AuthContextProvider>
+          <App />
+        </AuthContextProvider>
       </FilterContextProvider>
     </VideoContextProvider>
```
Comment (tusharkandpal) [inline src/index.js]: can define a single context provider to wrap up all other contexts & directly import & use here.

**Review by tusharkandpal — APPROVED** [review]
code is clean & looks good to merge.

 suggested few changes, can go through them.

File: src/Screen/LoginScreen/LoginScreen.jsx (hunk 2 above)
Comment (munjalSaksham71) [reply src/Screen/LoginScreen/LoginScreen.jsx:32]: this ```loginUser``` is  already in context.

**Pull request merged**